src/test/dokan/dokan.cc

```diff
@@ -0,0 +1,450 @@
+        0,             // use default security descriptor
+        CREATE_NEW,
+        FILE_ATTRIBUTE_NORMAL | FILE_FLAG_DELETE_ON_CLOSE,
+        0);
+
+    ASSERT_NE(hFile, INVALID_HANDLE_VALUE)
+        << "Could not open file: "
+        << DEFAULT_MOUNTPOINT"test_create.txt "
+        << "err: " << GetLastError() << std::endl;
+    
+    LARGE_INTEGER li;
+    li.QuadPart = 2097152; // 2MB
+    FILE_ALLOCATION_INFO fai;
+    fai.AllocationSize = li;
+    
+    ASSERT_NE(SetFileInformationByHandle(
+                hFile,
+                FileAllocationInfo,
+                &fai,
+                sizeof(FILE_ALLOCATION_INFO)
+             ),0) << "Error: " << GetLastError();
+
+    ASSERT_NE(CloseHandle(hFile), 0);
+    
+    // FILE_FLAG_DELETE_ON_CLOSE is used
+    ASSERT_FALSE(fs::exists(file_path));
+}
+
+TEST_F(DokanTests, test_file_type) {
+    std::string test_dir = DEFAULT_MOUNTPOINT"test_info_"
+                            + get_uuid() + "\\";
+    std::string file_path = test_dir + "file_"
+                            + get_uuid();
+    std::string dir_path = test_dir + "dir_"
+                           + get_uuid() + "\\";
+
+    ASSERT_TRUE(fs::create_directory(test_dir));
+
+    std::ofstream{file_path};
+    ASSERT_TRUE(fs::create_directory(dir_path));
+
+    ASSERT_TRUE(fs::is_regular_file(fs::status(file_path)));
+    ASSERT_TRUE(fs::is_directory(fs::status(dir_path)));
+
+    // clean-up
+    fs::remove_all(test_dir);
+
+}
+
+TEST_F(DokanTests, test_volume_info) {
+    // + 1 is for NULL
+    char volume_name[MAX_PATH + 1] = { 0 };
+    char file_system_name[MAX_PATH + 1] = { 0 };
+    DWORD serial_number = 0;
+    DWORD max_component_len = 0;
+    DWORD file_system_flags = 0;
+
+    ASSERT_EQ(
+        GetVolumeInformation(
+            DEFAULT_MOUNTPOINT,
+            volume_name,
+            sizeof(volume_name),
+            &serial_number,
+            &max_component_len,
+            &file_system_flags,
+            file_system_name,
+            sizeof(file_system_name)),TRUE) 
+        << "GetVolumeInformation() failed, error: "
+        << GetLastError() << std::endl;
+
+    ASSERT_STREQ(volume_name, "TestCeph") 
+        << "Received: " << volume_name << std::endl;
+    ASSERT_STREQ(file_system_name, "Ceph")
+        << "Received: " << file_system_name << std::endl;
+    ASSERT_EQ(max_component_len, 256);
+    ASSERT_EQ(serial_number, 1234567890)
+        << "Received: " << serial_number << std::endl;
+
+    // Consider adding specific flags 
+    // and check for them
+    // ASSERT_EQ(file_system_flags, 271);
+}
+
+TEST_F(DokanTests, test_get_free_space) {
+    std::error_code ec;
+    const std::filesystem::space_info si = std::filesystem::space(DEFAULT_MOUNTPOINT, ec);
+    
+    ASSERT_NE(static_cast<std::intmax_t>(si.capacity), 0);
+    ASSERT_NE(static_cast<std::intmax_t>(si.free), 0);
+    ASSERT_NE(static_cast<std::intmax_t>(si.available), 0);
+    ASSERT_EQ(ec.value(), 0);
+}
```

Comment:
There are a few other things that we should test:

* file timestamps
* disk free space - let's just ensure that we can retrieve those values and that after writing a file, TotalNumberOfBytes is greater or equal to that file size.
* reported file size
* CreateFile CreateDisposition tests